### DISCUSSION TITLE
nx monorepo and serwist

### DISCUSSION THREAD

**Discussion by bart-sofomo** [serwist/serwist#97] [Q&A]
Hi,
I'm struggling to get service worker working in vercel. I'm using nx workspace monorepo and my app is in `apps/my-app`.
The setup for serwist is like:
```
const withSerwist = withSerwistInit.default({
  swSrc: './app/sw.ts',
  swDest: './public/sw.js',
  swUrl: './sw.js',
  disable: process.env.NODE_ENV !== 'production',
});
```
On vercel it tries to reach `https://preview_url.app/apps/my-app/public/sw.js` which is wrong. It should be `https://preview_url.app/sw.js`.

Is there something wrong with my options? How can I make it working?

**Comment by DuCanhGH** [comment]
@bart-sofomo would you mind sharing a reproduction or editing Serwist's code so that it logs `basePath` upon initialization?

**Reply by bart-sofomo** [reply]
@DuCanhGH unfortunately I cannot share the code. what I can see when I hit `https://preview_url.app/sw.js` that inside I see (in `precacheEntries`):
```
[{'revision':'71782f0a19e1ac2b7126fa129afe790b','url':'/_next/../../../../apps/my-app/public/sw.js'},{'revision':'6524c3cb055a7a72ca835ea9fe90e591','url':'/_next/../../../../apps/my-app/public/sw.js.map'},
``` 

so basically the `sw.js` is populated under `https://preview_url.app/sw.js`.

Can you point me on how can I log `basePath` - I'm not sure how/where to do that?

**Reply by DuCanhGH** [reply]
@bart-sofomo ah. This dreaded bug again. Would you mind temporarily upgrading to Serwist preview? This has been fixed there. Should probably take a week before this gets to stable as well.

**Reply by bart-sofomo** [reply]
@DuCanhGH got it working with `^9.0.0-preview.12`
